.github/workflows/regression.yml

```diff
@@ -47,96 +45,77 @@ env:
-          elif [[ "${{ github.event_name }}" != "pull_request_review" &&
-                  "${{ !contains(github.event.pull_request.labels.*.name, 'ci-condition: skip-regression') }}" == "true" ]]; then
-            # The small caveat here is that we still want to run the suite via a review comment.
-            echo "SHOULD_RUN=false" >> $GITHUB_OUTPUT
-          else
-            echo "SHOULD_RUN=true" >> $GITHUB_OUTPUT
-          fi
-
   # Only run this workflow if files changed in areas that could possibly introduce a regression
-  check-source-changed:
+  should-run:
     runs-on: ubuntu-latest
     timeout-minutes: 5
-    needs: check-should-run
-    if: ${{ needs.check-should-run.outputs.should-run == 'true' }}
+    if: github.event_name != 'pull_request'
     outputs:
       source_changed: ${{ steps.filter.outputs.SOURCE_CHANGED }}
+      comment_valid: ${{ steps.comment.outputs.isTeamMember }}
     steps:
-    - uses: actions/checkout@v3
-
-    - name: Collect file changes
-      id: changes
-      if: github.event_name == 'merge_group'
-      uses: dorny/paths-filter@v3
-      with:
-        base: ${{ github.event.merge_group.base_ref }}
-        ref: ${{ github.event.merge_group.head_ref }}
-        list-files: shell
-        filters: |
-          all_changed:
-            - added|deleted|modified: "**"
-          ignore:
-            - "./.github/!(regression.yml)"
-            - "./.gitignore"
-            - "distribution/**"
-            - "rust-doc/**"
-            - "docs/**"
-            - "rfcs/**"
-            - "testing/**"
-            - "tilt/**"
-            - "website/**"
-            - "*.md"
-            - "Tiltfile"
-            - "netlify.toml"
-            - "NOTICE"
-            - "LICENSE-3rdparty.csv"
-            - "LICENSE"
-            - "lib/codecs/tests/data/**"
-
-    # This step allows us to conservatively run the tests if we added a new
-    # file or directory for source code, but forgot to add it to this workflow.
-    # Instead, we may unnecessarily run the test on new file or dir additions that
-    # wouldn't likely introduce regressions.
-    - name: Determine if should not run due to irrelevant file changes
-      id: filter
-      if: github.event_name == 'merge_group'
-      env:
-        ALL: ${{ steps.changes.outputs.all_changed_files }}
-        IGNORE: ${{ steps.changes.outputs.ignore_files }}
-      run: |
-        echo "ALL='${{ env.ALL }}'"
-        echo "IGNORE='${{ env.IGNORE }}'"
-        export SOURCE_CHANGED=$(comm -2 -3 <(printf "%s\n" "${{ env.ALL }}") <(printf "%s\n" "${{ env.IGNORE }}"))
-        echo "SOURCE_CHANGED='${SOURCE_CHANGED}'"
-
-        if [ "${SOURCE_CHANGED}" == "" ]; then
-          export SOURCE_CHANGED="false"
-        else
-          export SOURCE_CHANGED="true"
-        fi
-
-        echo "SOURCE_CHANGED='${SOURCE_CHANGED}'"
-        echo "SOURCE_CHANGED=${SOURCE_CHANGED}" >> $GITHUB_OUTPUT
+      - uses: actions/checkout@v3
```

Comment:
I recommend turning on the "hide whitespace" setting and reloading for an easier review.